src/providers/AgreementSubscription.js

```diff
@@ -0,0 +1,172 @@
+import React, { useContext, useMemo, useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import connectAgreement from '@aragon/connect-agreement'
+import { captureErrorWithSentry } from '../sentry'
+import { createAppHook, useApp } from '@aragon/connect-react'
+import { connector } from '../current-environment'
+import { useWallet } from '../providers/Wallet'
+import { useMounted } from '../hooks/useMounted'
+
+const { agreement } = connector
+
+const useAgreement = createAppHook(connectAgreement, agreement.connectorConfig)
+const AgreementSubscriptionContext = React.createContext()
+
+function AgreementSubscriptionProvider({ children }) {
+  const { account } = useWallet()
+  const [agreementApp, agreementAppStatus] = useApp(agreement.name)
+  const [currentVersion, currentVersionStatus] = useAgreement(
+    agreementApp,
+    (app) => app.onCurrentVersion()
+  )
+  const [disputableApps, disputableAppsStatus] = useAgreement(
+    agreementApp,
+    (app) => app.onDisputableApps()
+  )
+  const [stakingFactory, stakingFactoryStatus] = useAgreement(
+    agreementApp,
+    (app) => app.stakingFactory()
+  )
+  const [signer, signerStatus] = useAgreement(
+    agreementApp,
+    (app) => (account ? app.onSigner(account) : null),
+    [account]
+  )
```

Comment:
This code is very verbose but required because of the way the react library currently works. It's something I've been discussing with Pierre (among other things related to react api) and as we zone in on the refined next version we can come back here to tidy it up.